rotkehlchen/chain/ethereum/modules/ens/decoder.py

```diff
@@ -170,52 +185,58 @@ def _decode_ens_registry_with_fallback_event(self, context: DecoderContext) -> D
-                msg = str(e)
-                if isinstance(e, KeyError):
-                    msg = f'missing key {msg}'
-
-                log.debug(f'Failed to decode ENS set-text event due to {msg}')
-                return DEFAULT_DECODING_OUTPUT
-
-            node = context.tx_log.topics[1]
-            try:
-                address = self.ethereum.contracts.contract('ENS_PUBLIC_RESOLVER_2').call(
-                    node_inquirer=self.ethereum,
-                    method_name='addr',
-                    arguments=[node],
-                )
-                address = to_checksum_address(address)
-                ens_mapping = find_ens_mappings(
-                    ethereum_inquirer=self.ethereum,
-                    addresses=[address],
-                    ignore_cache=False,
-                )
-            except RemoteError as e:
-                log.debug(f'Failed to decode ENS set-text event due to {str(e)}')
-                return DEFAULT_DECODING_OUTPUT
+        if context.tx_log.topics[0] not in (TEXT_CHANGED_1, TEXT_CHANGED_2):
+            return DEFAULT_DECODING_OUTPUT
 
-            name_to_show = ens_mapping.get(address, address)
+        try:
+            _, decoded_data = decode_event_data_abi_str(
+                context.tx_log,
+                TEXT_CHANGED_1_ABI if context.tx_log.topics[0] == TEXT_CHANGED_1 else TEXT_CHANGED_2_ABI,  # noqa: E501
+            )
+            changed_key = decoded_data[0]
+            new_value = decoded_data[1] if context.tx_log.topics[0] == TEXT_CHANGED_2 else None  # noqa: E501
+        except DeserializationError as e:
+            log.error(f'Failed to decode ENS set-text event due to {str(e)}')
+            return DEFAULT_DECODING_OUTPUT
 
-            notes = f'Set ENS {changed_key} attribute for {name_to_show}'
-            context.decoded_events.append(self.base.make_event_from_transaction(
-                transaction=context.transaction,
-                tx_log=context.tx_log,
-                event_type=HistoryEventType.INFORMATIONAL,
-                event_subtype=HistoryEventSubType.NONE,
-                asset=A_ETH,
-                balance=Balance(),
-                location_label=context.transaction.from_address,
-                notes=notes,
-                counterparty=CPT_ENS,
-                address=context.transaction.to_address,
-            ))
+        node = context.tx_log.topics[1]  # node is a hash of the name used by ens internals
+        try:
+            with GlobalDBHandler().conn.read_ctx() as cursor:
+                contract = self.ethereum.contracts.contract_by_address(
+                    cursor=cursor,
+                    address=context.tx_log.address,  # Is either public resolver v2 or v3
+                )
+            address = contract.call(  # type: ignore[union-attr]  # the contract has to exist
+                node_inquirer=self.ethereum,
+                method_name='addr',
+                arguments=[node],
+            )
```

Comment:
this type ignore again I believe is trusting the developer too much. If for some reason the globaldb is modified, the code to get it from the database, the address changes due to us adding another decoding rule or something similar this can break. Can you add an if statament

```
if contract is None:
    log.error('Could not find ens contract...')
    return DEFAULT_DECODING_OUTPUT
```

I believe it should be even a user warning since is something critical if if fails and shouldn't go silent